### ISSUE TITLE
Can't compile the example app exploriak

### ISSUE THREAD

**Issue by schaary** [drewkerrigan/riak-elixir-client#51]
When I try to compile the example app exploriak, I get the error message

```bash
rm -f exploriak
mix deps.get
Running dependency resolution...
* Getting riak (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riak-1.0.0.tar)
  Using locally cached package
* Getting linguist (Hex package)
  Checking package (https://repo.hex.pm/tarballs/linguist-0.1.5.tar)
  Using locally cached package
* Getting pooler (Hex package)
  Checking package (https://repo.hex.pm/tarballs/pooler-1.5.0.tar)
  Using locally cached package
* Getting riakc (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riakc-2.1.1.tar)
  Using locally cached package
* Getting riak_pb (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riak_pb-2.1.0.tar)
  Using locally cached package
* Getting hamcrest (Hex package)
  Checking package (https://repo.hex.pm/tarballs/hamcrest-0.1.2.tar)
  Using locally cached package
* Getting protobuffs (Hex package)
  Checking package (https://repo.hex.pm/tarballs/protobuffs-0.8.2.tar)
  Using locally cached package
* Getting meck (Hex package)
  Checking package (https://repo.hex.pm/tarballs/meck-0.8.3.tar)
  Using locally cached package
mix escript.build
==> pooler (compile)
Compiled src/pooler_starter.erl
Compiled src/pooler_starter_sup.erl
Compiled src/pooler_sup.erl
Compiled src/pooler_pooled_worker_sup.erl
Compiled src/pooler_config.erl
Compiled src/pooler_app.erl
Compiled src/pooler_pool_sup.erl
Compiled src/pooler.erl
==> hamcrest (compile)
Compiled src/hamcrest.erl
Compiled src/hamcrest_term.erl
Compiled src/hamcrest_matchers.erl
==> hamcrest (post_compile)
==> linguist
Compiling 3 files (.ex)
warning: parentheses are required when piping into a function call. For example:

    foo 1 |> bar 2 |> baz 3

is ambiguous and should be written as

    foo(1) |> bar(2) |> baz(3)

Ambiguous pipe found at:
  lib/linguist/compiler.ex:72

Generated linguist app
==> meck (compile)
Compiled src/meck_util.erl
Compiled src/meck_ret_spec.erl
Compiled src/meck_proc.erl
Compiled src/meck_matcher.erl
Compiled src/meck_history.erl
Compiled src/meck_expect.erl
Compiled src/meck_cover.erl
Compiled src/meck_code.erl
Compiled src/meck_args_matcher.erl
Compiled src/meck_code_gen.erl
Compiled src/meck.erl
==> protobuffs (compile)
Compiled src/protobuffs_file.erl
Compiled src/protobuffs_scanner.erl
Compiled src/protobuffs_parser.erl
Compiled src/protobuffs_cli.erl
Compiled src/pokemon_pb.erl
Compiled src/protobuffs.erl
Compiled src/protobuffs_compile.erl
==> riak_pb (pre_compile)
==> riak_pb (compile)
Compiling src/riak_yokozuna.proto

=INFO REPORT==== 2-Jan-2017::13:33:45 ===
Writing header file to "riak_yokozuna_pb.hrl"

=INFO REPORT==== 2-Jan-2017::13:33:45 ===
Writing beam file to "riak_yokozuna_pb.beam"
Compiling src/riak_search.proto

=INFO REPORT==== 2-Jan-2017::13:33:45 ===
Writing header file to "riak_search_pb.hrl"

=INFO REPORT==== 2-Jan-2017::13:33:45 ===
Writing beam file to "riak_search_pb.beam"
Compiling src/riak_kv.proto

=INFO REPORT==== 2-Jan-2017::13:33:45 ===
Writing header file to "riak_kv_pb.hrl"

=INFO REPORT==== 2-Jan-2017::13:33:46 ===
Writing beam file to "riak_kv_pb.beam"
Compiling src/riak_dt.proto

=INFO REPORT==== 2-Jan-2017::13:33:46 ===
Writing header file to "riak_dt_pb.hrl"

=INFO REPORT==== 2-Jan-2017::13:33:46 ===
Writing beam file to "riak_dt_pb.beam"
Compiling src/riak.proto

=INFO REPORT==== 2-Jan-2017::13:33:46 ===
Writing header file to "riak_pb.hrl"

=INFO REPORT==== 2-Jan-2017::13:33:46 ===
Writing beam file to "riak_pb.beam"
Compiled src/riak_pb_kv_codec.erl
Compiled src/riak_pb_codec.erl
Compiled src/riak_pb_search_codec.erl
Compiled src/riak_pb_messages.erl
Compiled src/riak_pb_dt_codec.erl
==> riakc (compile)
ERROR: OTP release 19 does not match required regex R15|R16|17|18
ERROR: compile failed while processing /Users/schaary/code/riak-elixir-client/examples/exploriak/deps/riakc: rebar_abort
==> exploriak
** (Mix) Could not compile dependency :riakc, "/Users/schaary/.mix/rebar compile skip_deps=true deps_dir="/Users/schaary/code/riak-elixir-client/examples/exploriak/_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile riakc", update it with "mix deps.update riakc" or clean it with "mix deps.clean riakc"
make: *** [all] Error 1```

I run Elixir v1.3.4 with Erlang/OTP 19 on macOS 10.12.2

**Comment by schaary** [comment]
On Linux (Ubuntu 16.04 LTS) with Elixir v1.3.4 and Erlang/OTP 19 get the error

```bash
rm -f exploriak
mix deps.get
Running dependency resolution
A new Hex version is available (0.14.1 < 0.15.0), please update with `mix local.hex`
* Getting riak (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riak-1.0.0.tar)
  Fetched package
* Getting linguist (Hex package)
  Checking package (https://repo.hex.pm/tarballs/linguist-0.1.5.tar)
  Fetched package
* Getting pooler (Hex package)
  Checking package (https://repo.hex.pm/tarballs/pooler-1.5.0.tar)
  Fetched package
* Getting riakc (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riakc-2.1.1.tar)
  Fetched package
* Getting riak_pb (Hex package)
  Checking package (https://repo.hex.pm/tarballs/riak_pb-2.1.0.tar)
  Fetched package
* Getting hamcrest (Hex package)
  Checking package (https://repo.hex.pm/tarballs/hamcrest-0.1.1.tar)
  Fetched package
* Getting protobuffs (Hex package)
  Checking package (https://repo.hex.pm/tarballs/protobuffs-0.8.2.tar)
  Fetched package
* Getting meck (Hex package)
  Checking package (https://repo.hex.pm/tarballs/meck-0.8.3.tar)
  Fetched package
mix escript.build
==> hamcrest (compile)
src/hamcrest.erl:59: syntax error before: '/'
src/hamcrest.erl:63: syntax error before: '/'
src/hamcrest.erl:67: syntax error before: '/'
src/hamcrest.erl:72: syntax error before: '/'
src/hamcrest.erl:79: syntax error before: '/'
src/hamcrest.erl:90: syntax error before: '/'
src/hamcrest.erl:108: syntax error before: '/'
Compiling src/hamcrest.erl failed:
ERROR: compile failed while processing /home/schaary/code/riak-elixir-client/examples/exploriak/deps/hamcrest: rebar_abort
** (Mix) Could not compile dependency :hamcrest, "/home/schaary/.mix/rebar compile skip_deps=true deps_dir="/home/schaary/code/riak-elixir-client/examples/exploriak/_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile hamcrest", update it with "mix deps.update hamcrest" or clean it with "mix deps.clean hamcrest"
Makefile:4: recipe for target 'all' failed
make: *** [all] Error 1
```

**Comment by randysecrist** [comment]
Recently fixed a ton of compile warnings and errors.  Should be good to go under OTP20.
